### PR TITLE
Frontend Service Layer Refactor

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR commonizes and refactors the service layer down to one commonFormService.
- Change getAllSubmissionData to accept params object as argument
- Update commonFormService unit tests
- Generalize getDashboardSettings to getNamedSetting service call
- Add AppSettings constants
- Commonize getTypes service call
- Invert getAllSubmissionData to default to metadata only
- Add removeSubmission, updateStatusCodes and updateSubmission service calls
- Commonize sendSubmission service call
- Remove minesOperatorScreeningService
- Commonize getSubmission service call
- Commonize and parameterize getAllSubmissionData service call
- Update commonFormService to leverage isValidForm
- Add isValidForm utility function to constants
- Commonize getNotes and addNote service calls
- Update notes, settings and status service endpoints
- Move AdminNavBar into common admin folder
- Recluster unit tests into describe categories
- Update formName to use constant instead of route for team and requestreceipt Add RequestReceipt unit test

[](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1279)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

- Will need to pass through all vue components and change to try/catch format at some point

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-vmvfjv-tools.pathfinder.gov.bc.ca/dashboard?id=common-forms-toolkit-pr-101

File: app/frontend/src/services/commonFormService.js [hunk 1]
```diff
@@ -1,7 +1,231 @@
...
+  //
+
+  /**
+   * @function getAllSubmissionData
+   * Fetch the contents of all attestation submission metadata
+   * @param {string} form The form name
+   * @param {boolean} [complete=false] Request complete metadata
+   * @returns {Promise} An axios response
+   */
+  getAllSubmissionData(form, complete = false) {
```
Comment (parc-jason) [inline app/frontend/src/services/commonFormService.js:168]: Do you want to simplify this? 
just use tiny as the parameter and pass that into the BE?

File: app/frontend/src/services/commonFormService.js [hunk 2]
```diff
@@ -48,19 +279,7 @@ export default {
    * @returns {Promise} An axios response
    */
   requestTeamAccess(form) {
+    if (!isValidForm(form)) return Promise.reject('Invalid form specified');
```
Comment (parc-jason) [inline app/frontend/src/services/commonFormService.js:282]: this is called a lot.  can't we just write a wrapper around getting appAxios?
getAxios(form) that will do the check and reject?  something like that, have it done and handled in one place.

Comment (parc-jason) [reply app/frontend/src/services/commonFormService.js:282]: could even do it with like a service factory, pass the form name to the service factory and reject it immediately.  few ways to slice it and reduce the repetition.

Comment (loneil) [reply app/frontend/src/services/commonFormService.js:282]: Yeah having this same line everywhere must be something we can get around...

File: app/frontend/src/services/commonFormService.js (hunk 1 above)
Comment (jujaga) [reply app/frontend/src/services/commonFormService.js:170]: I just looked at this function again and inverted its default. It's now setting tiny=true by default, and only drops that parameter if and only if complete is set to true (aka I want all the data, not just metadata).

Comment (jujaga) [reply app/frontend/src/services/commonFormService.js:170]: Updated as discussed - passing in params by default instead for more caller flexibility.

File: app/frontend/src/services/commonFormService.js (hunk 2 above)
Comment (jujaga) [reply app/frontend/src/services/commonFormService.js]: I've updated this by wrapping appAxios function and throwing if invalid.

File: app/frontend/src/services/commonFormService.js (hunk 1 above)
Comment (parc-jason) [reply app/frontend/src/services/commonFormService.js]: Updating the server side to have deleted=false and tiny=true by default on submission search.

**Pull request merged**